project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,171 @@
+              if (triggersBuildAll)
+                log.info(s"GitHub PR comment [ ${c.getUrl} ] contains [$buildAllMagicPhrase], forcing BUILD ALL mode!")
+              triggersBuildAll
+            }
+          } catch {
+            case ex: Exception =>
+              log.warn("Unable to reach GitHub! Exception was: " + ex.getMessage)
+              false
+          }
+      }
+    },
+
+    projectIsAffectedByChanges in ValidatePR := (githubEnforcedBuildAll in ValidatePR).value || {
+      val log = streams.value.log
+      log.debug(s"Analysing project (for inclusion in PR validation): [${name.value}]")
+
+      // if in any scope, any of the changed modules is within this projects dependencies, we must test it:
+      val shouldBeBuilt = (changedDirectories in ValidatePR).value.exists { modifiedProject ⇒
+        Set(Compile, Test, Runtime, Provided, Optional) exists { ivyScope: sbt.Configuration ⇒
+          log.debug(s"Analysing [$ivyScope] scoped dependencies...")
+
+          def moduleId(artifactName: String) = ModuleId("com.typesafe.akka", artifactName + "_" + scalaBinaryVersion.value, version.value)
+          val modifiedModuleIds = Set(moduleId(modifiedProject), moduleId(modifiedProject + "-experimental"))
+
+          def resolutionFilename(includeScalaVersion: Boolean) =
+            s"%s-%s-%s.xml".format(
+              organization.value,
+              name.value + (if (includeScalaVersion) "_" + scalaBinaryVersion.value else ""),
+              ivyScope.toString())
+          def resolutionFile(includeScalaVersion: Boolean) =
+            target.value / "resolution-cache" / "reports" / resolutionFilename(includeScalaVersion)
+
+          val ivyReportFile = {
+            val f1 = resolutionFile(includeScalaVersion = true)
+            val f2 = resolutionFile(includeScalaVersion = false)
+            if (f1.exists()) f1 else f2
+          }
+
+          val deps = IvyGraphMLDependencies.graph(ivyReportFile.getAbsolutePath)
+          deps.nodes.foreach { m ⇒ log.debug(" -> " + m.id) }
+
+          // if this project depends on a modified module, we must test it
+          deps.nodes.exists { m =>
+            val depends = modifiedModuleIds exists { _.name == m.id.name } // match just by name, we'd rather include too much than too little
+            if (depends) log.info(s"Project [${name.value}] must be verified, because depends on [${modifiedModuleIds.find(_ == m.id).get}]")
+            depends
+          }
+        }
+      }
+
+      shouldBeBuilt
+    },
+
+    validatePullRequest := Def.taskDyn {
+      val log = streams.value.log
+      val theVoid = Def.task { () } // when you stare into the void, the void stares back at you
+
+      if ((projectIsAffectedByChanges in ValidatePR).value) {
+        log.info(s"Changes in PR are affecting project [${name.value}] - proceeding with test:test")
+        theVoid.dependsOn(test in Test)
```

Comment:
This should be `test in ValidatePR` (and the log string above `pr-validation:test`) as otherwise tag filters which are set on `ValidatePR` scope will not kick in.